### PR TITLE
Patch VTK to enable python 3.8 in VTK 8.2

### PR DESCRIPTION
**Pull request by kwryankrattiger**
Add VTK 8.2.1a, a patched VTK for Python 3.8 support.

Re-enable VisIt in the Data and Vis SDK CI.

Roughly based on patches from, which do not cleanly apply to 8.2.0, and some additional changes required for compilation.
https://gitlab.kitware.com/vtk/vtk/-/merge_requests/6269
https://gitlab.kitware.com/vtk/vtk/-/merge_requests/6275

CC: @tgamblin

### CODE REVIEW

**Comment by kwryankrattiger** [comment]
Ref. https://discourse.vtk.org/t/python-3-8-wrapping-support/2621/4 for additional context.

**Comment by kwryankrattiger** [comment]
@spackbot fix style

**Comment by kwryankrattiger** [comment]
Followup for: https://github.com/spack/spack/pull/38619

CC: @cyrush

**Comment by eugeneswalker** [comment]
Did you mean to make changes to `share/spack/gitlab/cloud_pipelines/configs/packages.yaml` here?

**Comment by kwryankrattiger** [comment]
> Did you mean to make changes to `share/spack/gitlab/cloud_pipelines/configs/packages.yaml` here?

Yes. Without that change it was concretizing the deprecated package in generate and then failing to build the deprecated python later.

**Comment by kwryankrattiger** [comment]
Testing offline as well here, concretize seems to be working as expected but still need to verify the VisIt 3.3.3 is working working with python 3.8.

@brugger1 I saw a comment in the Visit config from VisIt 3.2 that there was an issue with python 3.8. Is there more detail on that?

https://gitlab.spack.io/rkratt/pipeline-experiments/-/pipelines/464680

**Comment by brugger1** [comment]
@kwryankrattiger I don't have a gitlab account to follow that link. Can you paste the relevant portion in this PR?

**Review by cyrush — DISMISSED** [review]

**Comment by eugeneswalker** [comment]
The three `vtk@8.2.1a` builds are failing in ECP DAV CI stack:
* [vtk/eyzjquq 8.2.1a gcc@=11.1.0 linux-ubuntu20.04-x86_64_v3 Data and Vis SDK](https://gitlab.spack.io/spack/spack/-/jobs/7909659)
```
...
1 error found in build log:
     232      CMake/vtkMPI.cmake:3 (find_package)
     233      CMake/FindHDF5.cmake:9 (include)
     234      CMake/vtkModuleMacros.cmake:891 (find_package)
     235      ThirdParty/hdf5/CMakeLists.txt:5 (vtk_module_third_party)
     236    This warning is for project developers.  Use -Wno-dev to suppress i
            t.
     237    
  >> 238    CMake Error at CMake/vtkModuleMacros.cmake:891 (find_package):
     239      By not providing "Findpugixml.cmake" in CMAKE_MODULE_PATH this pr
            oject has
     240      asked CMake to find a package configuration file provided by "pug
            ixml", but
     241      CMake did not find one.
     242    
     243      Could not find a package configuration file provided by "pugixml"
             with any
     244      of the following names:
```

**Comment by kwryankrattiger** [comment]
@eugeneswalker 
I fixed this in a branch. I am not sure how VTK 8.2 was ever building because I didn't add the pugixml dep in my patch. I added it as a dependency in my testing branch and getting that to work first.

File: var/spack/repos/builtin/packages/paraview/package.py [hunk 1]
```diff
@@ -253,7 +253,7 @@ class Paraview(CMakePackage, CudaPackage, ROCmPackage):
 
     # ParaView depends on proj@8.1.0 due to changes in MR
```
Comment (danlipsa) [inline var/spack/repos/builtin/packages/paraview/package.py:254]: You may want to clarify this comment

**Comment by danlipsa** [comment]
Looks good after fixing the mentioned comment.

File: var/spack/repos/builtin/packages/paraview/package.py (hunk 1 above)
Comment (kwryankrattiger) [reply var/spack/repos/builtin/packages/paraview/package.py:254]: good call 👍🏽

**Review by eugeneswalker — APPROVED** [review]

**Pull request merged**